src/Components/MasterComponnets/AddOrderModal/AddOrderModal.tsx

```diff
@@ -0,0 +1,228 @@
+import {
+  Box,
+  Button,
+  FormControl,
+  Modal,
+  Stack,
+  Typography,
+} from "@mui/material";
+import React from "react";
+import { FormTextField } from "../../AuthComponents/FormTextField/FormTextField";
+import { useForm } from "react-hook-form";
+import ButtonForm from "../../AuthComponents/ButtonForm/ButtonForm";
+import { SquareCheckBig } from "lucide-react";
+
+const style = {
+  position: "absolute",
+  top: "50%",
+  left: "50%",
+  transform: "translate(-50%, -50%)",
+  width: 650,
+  bgcolor: "background.paper",
+  fontFamily: "inter",
+  py: 2.5,
+  px: 4,
+  borderRadius: "8px",
+};
+export default function AddOrderModal() {
+  const [open, setOpen] = React.useState(false);
+
+  const {
+    register,
+    handleSubmit,
+    setFocus,
+    formState: { errors, isSubmitting },
+    reset,
+  } = useForm();
+
+  const handleOpen = () => setOpen(true);
+  const handleClose = () => {
+    setOpen(false);
+    reset();
+  };
+
+  React.useEffect(() => {
+    setFocus("ProductName");
+  }, [setFocus]);
+
+  const onSubmit = (data) => {
+    console.log(data); // Handle the submitted data here
+  };
+
+  return (
+    <div>
+      <Button
+        sx={{
+          backgroundColor: "#1366D9",
+          color: "#FFFFFF",
+          padding: "10px 16px",
+          textTransform: "none",
+        }}
+        onClick={handleOpen}
+      >
+        Add Product
+      </Button>
+      <Modal
+        keepMounted
+        open={open}
+        onClose={handleClose}
+        aria-labelledby="keep-mounted-modal-title"
+        aria-describedby="keep-mounted-modal-description"
+      >
+        <Box sx={style}>
+          <Typography
+            variant="h6"
+            component={"h2"}
+            sx={{
+              color: "#383E49",
+              fontWeight: "500",
+              fontSize: "20px",
+            }}
+          >
+            New Order
+          </Typography>
+          <FormControl
+            component={"form"}
+            sx={{ mt: "20px", width: "100%" }}
+            onSubmit={handleSubmit(onSubmit)} // Updated here
+          >
+            <Stack spacing={1.7}>
+              <FormTextFieldWithLabel
+                label="Product Name"
+                name="Name"
+                placeholder="Enter Product Name"
+                register={register}
```

Comment:
If we created the component using  forwarded ref as we did in the Component API design, we will not need to pass the register, rules and name as sepearte props. We will be able to deal the the component as a normal TextField.
```
<FormTextFieldWithLabel {...register("productId",{required: "Product ID is required"})}/>
```